Victor9k Double Sided Support

pauldevine opened the pull request:
Hi David,
Thanks for your help earlier. Here's a cleaner re-take on Victor9k double sided disks. I spent a lot of time iterating through what seemed like a bug in the encoder/decoder but was really a demonstration of my lack of bit-shifting skills. 

One note, I'm making an intentional change with track 48. The documentation shows it in zone 4 with 15 sectors and a speed of 187.6 original_period_ms. It actually is in zone 5 with 14 sectors and a speed of 175.3. I'm basing that off the zone organization I'm seeing in the BIOS and the format.com code. The mame code also makes this same correction to what's a bug in the documentation. I've reflected that change on the single sided disks also. 

This all seems to work fine when reading physical media to .img files and .flux files. I can then translate those files back and forth between each other in a reliable manner on my hard drive. I still, however, get errors when I try to write the .img file back out to a physical floppy. This happens on the master branch as well as the victor9k branch before my changes. I'll keep playing with that issue and will hopefully have a separate PR for you on that too.

davidgiven commented:
I'll take your word for it as you have the hardware and I don't! One nitpick: could you update the table in doc/disk-victor9k.md to match the correct layout, please? (And add a note that the documentation is wrong...)

What kind of errors are you getting when writing?

pauldevine commented:
Thanks for the catch on the docs miss. I've updated the documentation. I also caught a docs bug, the existing doc was referencing the format name as victor9k-ss (hyphen) instead of victor9k_ss (underscore). I'm presuming the code had the correct value and not the docs so I corrected it. 

Also, I should have mentioned, I figured out the diskcopy utility that shipped with the Victor has a flag to output diagnostic data while writing a disk. So I have confirmed the track values from that also. One part of the output that surprised me was the gap2 values it displayed. I saw one disk as low as 28 and then this one here has a track at 64. I had no idea it could vary that much.

![Victor 9000 Diskcopy Track Information](https://user-images.githubusercontent.com/1730609/150276443-ff1fb37f-d97c-4106-8617-02120cd903b6.jpeg)

In terms of writing issues, I'm having an issue when I try to write to physical media via a greaseweazel. The same thing is happening to me on master & on the victor9k branch. When I try to write to disk I get the following output:

```
(base) pauldevine@Pauls-MacBook-Air fluxengine % ./fluxengine write victor9k_ss -i msdos31_ss.img
IMG: read 80 tracks, 1 sides, 0 kB total
Using GreaseWeazle GW135195034205400002CD9705 on /dev/cu.usbmodem11501
Measuring rotational speed... 166ms
Writing to: drive 0
  0.0: Write:   166 ms in 62723 bytes
       Verify:  332 ms in 127345 bytes
       Rewrite: 166 ms in 62723 bytes
       Verify:  332 ms in 127797 bytes
       Rewrite: 166 ms in 62723 bytes
       Verify:  332 ms in 127680 bytes
       Rewrite: 166 ms in 62723 bytes
       Verify:  332 ms in 127751 bytes
       Rewrite: 166 ms in 62723 bytes
       Verify:  332 ms in 127602 bytes
       Rewrite: 166 ms in 62723 bytes
       Verify:  332 ms in 127428 bytes
Error: Write failed; uncorrectable error during write.
````
Looks like something is wrong in the expected byte array sizes. I spent some time trying to debug it and got really lost. I'm sorry I wasn't able to track it down. What's weird is this used to work. I couldn't get the resulting disk to be read by the victor but I did write several. To try and debug I checked out old git sha's, rebuilt and tried to get it to work and the older ones aren't working either. I'm confused as to what's happening.

davidgiven commented:
Love those green screens... I remember seeing a reference in the technical manual that the format command adapts the gap2 value by measuring some drive parameter, but it didn't go into much detail on what or why.

Re failing to write: I've just downloaded your branch and tried it myself, and it worked --- I wrote a disk and it verified fine. But I did have to merge with master to bring the branch up to date. It's possible you're missing some critical bugfixes? It looks like your branch is old. Could you try updating it? There should be a github button to do it for you on the code page...

pauldevine commented:
Thanks, rebasing to master did the trick. Not sure what that was. I'm writing correctly to physical media. 

IIRC the gap2 stuff was to accommodate slight differences in speed on various machines. I just didn't realize they would vary that much.

davidgiven commented:
Fantastic. Okay, merging.

So, you're reading and writing to media but the disks don't work in a real machine? That'll point at a bug in my encoder, then...
